File: core/src/main/java/javabot/model/OperationEvent.java
package javabot.model;

import org.mongodb.morphia.annotations.Entity;

@Entity("events")
public class OperationEvent extends AdminEvent {
    private String operation;

    public OperationEvent() {
    }

    public String getOperation() {
        return operation;
    }

    public void setOperation(String operation) {
        this.operation = operation;
    }

    @Override
    public void add() {
        getBot().enableOperation(operation);
    }

    @Override
    public void delete() {
        getBot().disableOperation(operation);
    }
}

File: src/main/kotlin/javabot/model/OperationEvent.kt
package javabot.model

import dev.morphia.annotations.Entity

@Entity("events")
class OperationEvent() : AdminEvent() {

    lateinit var operation: String

    constructor(requestedBy: String, type: EventType, operation: String): this() {
        this.requestedBy = requestedBy
        this.type = type
        this.operation = operation
    }

    override fun add() {
        bot.enableOperation(operation)
    }

    override fun delete() {
        bot.disableOperation(operation)
    }
}
